lib/pbench/test/functional/server/test_user.py

```diff
@@ -96,16 +147,36 @@ def test_update(self, pbench: Pbench):
         payload = user.copy()
         del payload["registered_on"]
 
-        response = pbench.put("user", {"target_username": self.USERNAME}, json=payload)
+        response = pbench_server_client.put(
+            "user", {"target_username": USERNAME1}, json=payload
+        )
+        put_response = response.json()
 
-        response = pbench.get("user", {"target_username": self.USERNAME})
+        response = pbench_server_client.get("user", {"target_username": USERNAME1})
         updated = response.json()
         assert user == updated
+        assert put_response == user
 
-    def test_delete(self, pbench: Pbench):
+    def test_delete(self, pbench_server_client: PbenchServerClient):
         """
         Test that we can delete our own user profile once logged in.
```

Comment:
This test description suggests that there ought to be two more tests:
- that we cannot delete our profile when _not_ logged in
- that we cannot delete _another_ user's profile.

I'm not sure that you're shooting for completeness in this PR, but it would be easy to do the first of those cases on `USERNAME2` before executing `test_cleanup_user2()`, and it would be easy to do the second on `USERNAME2` before executing `test_logout()`.

Reply:
Ah, yes; we know we can't *read*, but verifying we can't guess and delete is probably not an entirely bad idea.